app/src/main/java/com/alphawallet/app/widget/ActionSheet.java

```diff
@@ -0,0 +1,80 @@
+package com.alphawallet.app.widget;
+
+import android.content.Context;
+
+import androidx.activity.result.ActivityResult;
+import androidx.annotation.NonNull;
+
+import com.alphawallet.app.web3.entity.Web3Transaction;
+import com.google.android.material.bottomsheet.BottomSheetDialog;
+
+import java.math.BigInteger;
+
+/**
+ * Created by JB on 20/11/2022.
+ */
+public abstract class ActionSheet extends BottomSheetDialog
+{
+    public ActionSheet(@NonNull Context context)
+    {
+        super(context);
+    }
+
+    public void setCurrentGasIndex(ActivityResult result)
+    {
+        throw new RuntimeException("Implement setCurrentGasIndex");
```

Comment:
How about keep these methods `abstract`, so that developers who want to extend it can get errors in immediately other than runtime.